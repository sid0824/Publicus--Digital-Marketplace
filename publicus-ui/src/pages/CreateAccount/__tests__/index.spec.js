import toJson from 'enzyme-to-json';
import React from "react";
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { CreateAccount } from "../index";
import { Auth, API } from 'aws-amplify';

configure({ adapter: new Adapter() });

const mockValues = {
  email: 'MOCK_EMAIL',
  password: 'MOCK_PASSWORD',
  password2: 'MOCK_PASSWORD',
  firstname: 'MOCK_FIRST',
  lastname: 'MOCK_LAST',
  confirmationCode: 'MOCK_CONF_CODE'
};

const mockGetFieldDecorator = jest.fn();
const mockForm = {
  validateFields: jest.fn().mockImplementation(async (func) => func(null, mockValues)),
  getFieldDecorator: mockGetFieldDecorator,
  getFieldsValue: () => {
    return {
      password: 'PASS',
      password2: 'PASSWORD',
    }
  }
};

const mockPush = jest.fn();
const mockSetUserHasAuthenticated = jest.fn();
const mockHistory = {
  push: mockPush
};


describe('Create Account', () => {

  beforeEach(() => {
    mockGetFieldDecorator.mockReturnValue((component) => component);
    Auth.signUp = jest.fn();
    Auth.confirmSignUp = jest.fn();
    Auth.signIn = jest.fn();
    Auth.currentCredentials = jest.fn().mockResolvedValue({
      params: {
        IdentityId: 'MOCK_ID'
      }
    });
    API.post = jest.fn().mockResolvedValue(null);
  });

  afterEach(() => {
    mockPush.mockReset();
    mockSetUserHasAuthenticated.mockReset();
    mockGetFieldDecorator.mockReset();
    Auth.signUp.mockReset();
    Auth.confirmSignUp.mockReset();
    Auth.signIn.mockReset();
    API.post.mockReset();
  });

  test('should render the create account page', () => {
    const wrapper = shallow(
      <CreateAccount
        form={mockForm}
        history={mockHistory}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.setState({
      confirmingUser: true
    });
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('should create and confirm an account', async() => {
    const wrapper = shallow(
      <CreateAccount
        form={mockForm}
        history={mockHistory}
        setUserHasAuthenticated={mockSetUserHasAuthenticated}
      />
    );
    await wrapper.instance().onCreateAccount({ preventDefault: () => {}});
    expect(Auth.signUp.mock.calls[0]).toMatchSnapshot();
    expect(wrapper.state()).toMatchSnapshot();
    await wrapper.instance().onConfirmAccount({ preventDefault: () => {}});
    await Promise.resolve();
    expect(API.post.mock.calls[0]).toMatchSnapshot();
    expect(mockSetUserHasAuthenticated).toHaveBeenCalledWith(true);
  });

});