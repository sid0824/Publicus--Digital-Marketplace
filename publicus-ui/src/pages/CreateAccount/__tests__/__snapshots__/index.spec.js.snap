// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Account should create and confirm an account 1`] = `
Array [
  Object {
    "password": "MOCK_PASSWORD",
    "username": "MOCK_EMAIL",
  },
]
`;

exports[`Create Account should create and confirm an account 2`] = `
Object {
  "confirming": false,
  "confirmingUser": true,
  "creating": false,
  "firstname": "MOCK_FIRST",
  "lastname": "MOCK_LAST",
  "password": "MOCK_PASSWORD",
  "username": "MOCK_EMAIL",
}
`;

exports[`Create Account should create and confirm an account 3`] = `
Array [
  "prod-publicus-app-api",
  "/user",
  Object {
    "body": Object {
      "cognitoId": "MOCK_ID",
      "email": "MOCK_EMAIL",
      "firstname": "MOCK_FIRST",
      "lastname": "MOCK_LAST",
    },
  },
]
`;

exports[`Create Account should render the create account page 1`] = `
<div
  style={
    Object {
      "marginLeft": "15%",
      "marginRight": "15%",
      "marginTop": "70px",
      "width": "70%",
    }
  }
>
  <h1>
    Enter Your Details
  </h1>
  <Form
    colon={true}
    hideRequiredMark={false}
    layout="horizontal"
    onSubmit={[Function]}
  >
    Email Address
    <FormItem
      hasFeedback={false}
    >
      <Input
        placeholder="Enter your email address..."
        type="text"
      />
    </FormItem>
    First Name
    <FormItem
      hasFeedback={false}
    >
      <Input
        placeholder="Enter your first name..."
        type="text"
      />
    </FormItem>
    Last Name
    <FormItem
      hasFeedback={false}
    >
      <Input
        placeholder="Enter your last name..."
        type="text"
      />
    </FormItem>
    Password
    <FormItem
      hasFeedback={false}
    >
      <Input
        placeholder="Enter your password..."
        type="password"
      />
    </FormItem>
    Confirm Your Password
    <FormItem
      hasFeedback={false}
      help="Passwords Should Match"
      validateStatus="error"
    >
      <Input
        placeholder="Confirm your Password..."
        type="password"
      />
    </FormItem>
    <FormItem
      hasFeedback={false}
    >
      <Button
        block={false}
        ghost={false}
        htmlType="submit"
        loading={false}
        type="primary"
      >
        Submit
      </Button>
    </FormItem>
  </Form>
</div>
`;

exports[`Create Account should render the create account page 2`] = `
<div
  style={
    Object {
      "marginLeft": "15%",
      "marginRight": "15%",
      "marginTop": "70px",
      "width": "70%",
    }
  }
>
  <h1>
    Enter Your Details
  </h1>
  <Form
    colon={true}
    hideRequiredMark={false}
    layout="horizontal"
    onSubmit={[Function]}
  >
    Check your email for the confirmation code
    <FormItem
      hasFeedback={false}
    >
      <Input
        placeholder="Confirmation Code..."
        type="text"
      />
    </FormItem>
    <FormItem
      hasFeedback={false}
    >
      <Button
        block={false}
        ghost={false}
        htmlType="submit"
        loading={false}
        type="primary"
      >
        Submit
      </Button>
    </FormItem>
  </Form>
</div>
`;
